homeassistant/components/zeroconf/__init__.py

```diff
@@ -268,10 +268,28 @@ def service_update(zeroconf, service_type, name, state_change):
                     # likely bad homekit data
                     return
 
-        for domain in zeroconf_types[service_type]:
+        for entry in zeroconf_types[service_type]:
+            if len(entry) > 1:
+                if "macaddress" in entry:
+                    if "properties" not in info:
+                        continue
+                    if "macaddress" not in info["properties"]:
+                        continue
+                    if (
+                        not info["properties"]["macaddress"]
+                        .upper()
+                        .startswith(entry["macaddress"])
+                    ):
+                        continue
+                if "name" in entry:
+                    if "name" not in info:
+                        continue
+                    if not info["name"].lower().startswith(entry["name"]):
```

Comment:
I think it would be awesome if we could use regular expressions here...

Reply:
Regex is a bit much, maybe [fnmatch](https://docs.python.org/3/library/fnmatch.html) so we can write `Shelly*`

Reply:
will do

Reply:
Yeah fnmatch will do... I'm just a sucker for regular expressions 👍